shotover-proxy/src/message/mod.rs

```diff
@@ -40,6 +43,9 @@ pub struct Message {
     // TODO: Not a fan of this field and we could get rid of it by making TimestampTagger an implicit part of ConsistentScatter
     // This metadata field is only used for communication between transforms and should not be touched by sinks or sources
     pub meta_timestamp: Option<i64>,
+
+    /// State of connection that sent or received this message
+    pub(crate) connection_state: ConnectionState,
```

Comment:
Since the `Message` itself won't be modifying this field, we could probably have this as a reference to the server's connection state.